### PR TITLE
Fix regression in Visual Editor handling of raw html blocks

### PR DESCRIPTION
**Pull request by gtritchie**
### Intent

Addresses #15189

### Approach

Use custom pandoc writer script to preserve the raw HTML blocks. See https://github.com/quarto-dev/quarto/pull/553 for essentially the same change in the VSCode Quarto extension.

### Automated Tests

None - I will add some tests in `main` when I forward-port this change.

### QA Notes

The modified code is invoked whenever the visual editor saves an R Markdown or Quarto file and when you switch between the Visual Editor and source mode.
 
Test RMD and QMD files containing raw HTML blocks and inline HTML, and switch back and forth from source mode to visual mode. Ensure the structure is preserved, especially that the raw block delimiters don't vanish, and inline HTML doesn't get modified. There may be whitespace changes when switching modes, that isn't new. Here's an example doc:

````qmd
---
title: Raw HTML
---

## Raw Block

```{=html}
Hello <i>World</i>, how are you?
```

## Inline HTML

Some more text and some <b>Bold</b> stuff.

````

### Documentation

NA

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Comment by gtritchie** [comment]
Check build failure I need to investigate before we merge this, so changed to draft.

File: src/cpp/session/modules/panmirror/SessionPanmirrorPandoc.cpp [hunk 1]
```diff
@@ -95,6 +96,17 @@ void pandocAstToMarkdown(const json::JsonRpcRequest& request,
...
+   // https://github.com/rstudio/rstudio/issues/15189
+    boost::regex pattern(R"(markdown(\+|-|$))");
+   if (boost::regex_search(format, pattern)) {
+     FilePath wdWriterLua = FilePath(session::options().rResourcesPath()).completePath("panmirror-scripts/md-writer.lua");
+      if (wdWriterLua.exists())
+         format = boost::regex_replace(format, boost::regex("markdown"), wdWriterLua.getAbsolutePathNative());
+      else
+         LOG_ERROR_MESSAGE("Custom markdown writer not found at: '" + wdWriterLua.getAbsolutePathNative() + "'");
+   }
+
```
Comment (kevinushey) [inline src/cpp/session/modules/panmirror/SessionPanmirrorPandoc.cpp:109]: Nit: indentation looks off here?

**Review by kevinushey — APPROVED** [review]
LGTM!

File: src/cpp/session/modules/panmirror/SessionPanmirrorPandoc.cpp (hunk 1 above)
Comment (gtritchie) [reply src/cpp/session/modules/panmirror/SessionPanmirrorPandoc.cpp]: Yup, fixed!

**Pull request merged**